test/common/upstream/load_balancer_impl_test.cc

```diff
@@ -561,5 +563,39 @@ TEST_F(RandomLoadBalancerTest, Normal) {
   EXPECT_EQ(cluster_.healthy_hosts_[1], lb_.chooseHost(nullptr));
 }
 
+TEST(LoadBalancerSubsetInfoImplTest, DefaultConfigIsDiabled) {
+  auto subset_info =
+      LoadBalancerSubsetInfoImpl(envoy::api::v2::Cluster::LbSubsetConfig::default_instance());
+
+  EXPECT_FALSE(subset_info.isEnabled());
+  EXPECT_TRUE(subset_info.fallbackPolicy() == envoy::api::v2::Cluster::LbSubsetConfig::NO_FALLBACK);
+  EXPECT_EQ(subset_info.defaultSubset().fields_size(), 0);
+  EXPECT_EQ(subset_info.subsetKeys().size(), 0);
+}
+
+TEST(LoadBalancerSubsetInfoImplTest, SubsetConfig) {
+  auto subset_value = ProtobufWkt::Value();
+  subset_value.set_string_value("the value");
+
+  auto subset_selector = envoy::api::v2::Cluster::LbSubsetConfig::LbSubsetSelector();
+  subset_selector.add_keys("selector_key");
+
+  auto subset_config = envoy::api::v2::Cluster::LbSubsetConfig::default_instance();
+  subset_config.set_fallback_policy(envoy::api::v2::Cluster::LbSubsetConfig::DEFAULT_SUBSET);
+  subset_config.mutable_default_subset()->mutable_fields()->insert({"key", subset_value});
+  subset_config.mutable_subset_selectors()->Add()->CopyFrom(subset_selector);
```

Comment:
Nit: If you build this in the reverse direction (i.e. acquire the selector via subset_config.mutable_subset_selectors()->Add()), you don't have to copy.